addons/sourcemod/scripting/nd_discord_playerlist.sp

```diff
@@ -0,0 +1,193 @@
+        SaveTeamsTimer = CreateTimer(1.0, Timer_SaveTeams, _, TIMER_FLAG_NO_MAPCHANGE);
+        return Plugin_Continue;
+    }
+    SaveTeamsTimer = null;
+
+    if(!IsValidHandle(DatabaseHandle)) {
+        LogAction(0, -1, "Database not connected");
+        return Plugin_Continue;
+    }
+
+    SavingTeams = true;
+
+    char query[1024];
+    FormatEx(query, sizeof(query), "TRUNCATE TABLE players;");
+    SQL_TQuery(DatabaseHandle, SaveTeams_Step1_RemovePlayers, query);
+
+    return Plugin_Continue;
+}
+
+void SaveTeams_Step1_RemovePlayers(Handle owner, Handle handle, const char[] error, any:data)
+{
+    if (error[0] != '\0')
+    {
+        LogAction(0, -1, "Removing players failed: %s", error);
+        SavingTeams = false;
+        return;
+    }
+
+    int[] consortium = new int[MAXPLAYERS + 1];
+    int numConsortium = 0;
+    int[] empire = new int[MAXPLAYERS + 1];
+    int numEmpire = 0;
+    int[] spectator = new int[MAXPLAYERS + 1];
+    int numSpectator = 0;
+    for (int client = 1; client <= MaxClients; client++)
+    {
+        if (IsValidClient(client, false))
+        {
+            int team = GetClientTeam(client);
+            if (team == TEAM_CONSORT)
+            {
+                consortium[numConsortium] = client;
+                numConsortium++;
+            }
+            else if(team == TEAM_EMPIRE)
+            {
+                empire[numEmpire] = client;
+                numEmpire++;
+            }
+            else
+            {
+                spectator[numSpectator] = client;
+                numSpectator++;
+            }
+        }
+    }
+
+    // We're done if we have no players to add
+    if(numConsortium + numEmpire + numSpectator == 0) {
+        SavingTeams = false;
+        return;
+    }
+
+    char query[10000];
+    char queryvalues[10000];
+    char[][] queryparts = new char[numConsortium + numEmpire + numSpectator][1024];
+    int part = 0;
+    char escapedName[MAX_NAME_LENGTH];
+
+    for (int player = 0; player < numConsortium; player++)
+    {
+        SQL_EscapeString(DatabaseHandle, PlayerNames[consortium[player]], escapedName, sizeof(escapedName));
+        FormatEx(queryparts[part++], 1024, "('%s', '%s')", escapedName, "consortium");
+    }
+
+    for (int player = 0; player < numEmpire; player++)
+    {
+        SQL_EscapeString(DatabaseHandle, PlayerNames[empire[player]], escapedName, sizeof(escapedName));
+        FormatEx(queryparts[part++], 1024, "('%s', '%s')", escapedName, "empire");
+    }
+
+    for (int player = 0; player < numSpectator; player++)
+    {
+        SQL_EscapeString(DatabaseHandle, PlayerNames[spectator[player]], escapedName, sizeof(escapedName));
+        FormatEx(queryparts[part++], 1024, "('%s', '%s')", escapedName, "spectator");
+    }
+
+    ImplodeStrings(queryparts, numConsortium + numEmpire + numSpectator, ", ", queryvalues, sizeof(queryvalues));
+    FormatEx(query, sizeof(query), "INSERT INTO players (name, team) VALUES %s;", queryvalues);
```

Comment:
Semi-colon at the end should be removed before giving to SQL_TQuery.

Reply:
done